Generator/Sources/NeedleFramework/Parsing/Pluginized/Tasks/PluginizedASTDeclarationParserTask.swift

```diff
@@ -39,97 +39,110 @@ class PluginizedDeclarationsParserTask: AbstractTask<PluginizedDependencyGraphNo
     override func execute() throws -> PluginizedDependencyGraphNode {
         let baseTask = DeclarationsParserTask(ast: ast)
         let baseNode = try baseTask.execute()
-        let (pluginizedComponents, nonCoreComponents, pluginExtensions) = try parsePluginizedStructures()
+        let visitor = Visitor(sourceHash: ast.sourceHash)
+        visitor.walk(ast.sourceFileSyntax)
```

Comment:
There are 4 total `visitor.walk` calls, my guess is that 2/3 of them actually get called in a real run. 

Are we sure that the multiple calls don't add any performance overhead? Or were you planning to land this and look into folding them in at a later date?